### PR TITLE
Fix precedence exception bug for Spanish and French passive

### LINKED ISSUE #1429: Spanish participle precedence not being exempt from passive voice marker
## What did you expect to happen?

The Spanish participle precedence to be exempt from being marked.

## What happened instead?

It was marked as being a passive sentence.

## How can we reproduce this behavior?

* Use the sentence `Los niños que son famosos estaban en un programa de televisión muy esperado.` in your text.
* Check that is gets marked as being passive when clicking the marker button in the interface.

## Technical info

* Platform: WordPress
* Yoast SEO version: 1.3.0

### PR DESCRIPTION
**Pull request by manuelaugustin**
## Summary

This PR can be summarized in the following changelog entry: 

* Fixes a bug that broke a filter which marks Spanish and French sentences as non-passive when certain exception words occur between the auxiliary and the participle.

## Relevant technical choices:

* Adds `language` to the participle attributes.

## Test instructions

This PR can be tested by following these steps:

In the browserified stand-alone:
* Check out this branch. Dont forget to use `grunt build:js`
* Set the language to `es_ES`
* Paste this sentence: `Los niños que son famosos estaban en un programa de televisión muy esperado.`
* Make sure it is not marked as passive.

NB: This branch was also tested in WordPress SEO. 

Fixes #1429

### CODE REVIEW

File: src/stringProcessing/precedenceException.js [hunk 1]
```diff
@@ -18,14 +18,14 @@ var cannotBeBetweenAuxiliaryAndParticipleSpanish = require( "../researches/spani
  */
 module.exports = function( sentencePart, participleIndex ) {
 	var precedenceExceptionRegex;
-	switch ( this.constructor.name ) {
-		case "FrenchParticiple":
+	switch ( this.getLanguage() ) {
```
Comment (igorschoester) [inline src/stringProcessing/precedenceException.js]: The `this` is still being used here. While not being bound to any particular scope. I'd really want to make sure this is correct by passing along the language and executing `this.getLanguage` in the proper scope (Participle).

File: src/stringProcessing/directPrecedenceException.js [hunk 2]
```diff
@@ -17,14 +17,14 @@ var cannotDirectlyPrecedePassiveParticipleSpanish = require( "../researches/span
  */
 module.exports = function( sentencePart, participleIndex ) {
 	var directPrecedenceExceptionRegex;
-	switch ( this.constructor.name ) {
-		case "FrenchParticiple":
+	switch ( this.getLanguage() ) {
```
Comment (igorschoester) [inline src/stringProcessing/directPrecedenceException.js]: See precedenceException comment.

**Comment by afercia** [comment]
CR 👍 
Pushed a few minot coding standards (double spaces).
The only thing I'm not sure it's intentional or not is the committed `yarn.lock` file, seems to me there's no reason to commit it but that can be evaluated during acceptance.

**Comment by jcomack** [comment]
@afercia If something changes in your `yarn.lock` file (in this case, `node-sass` was updated to properly reflect the version present in `package.json`), you *always* want to commit it to prevent possible version conflicts between (developer) environments.

**Comment by jcomack** [comment]
Acceptance done 👍

**Pull request merged**